src/configuration.ts

```diff
@@ -128,15 +129,46 @@ export function loadConfigurationFromPath(configFilePath: string): IConfiguratio
         console.warn(PACKAGE_DEPRECATION_MSG);
         return require(configFilePath).tslintConfig;
     } else {
-        let fileData = fs.readFileSync(configFilePath, "utf8");
-        // remove BOM if present
-        fileData = fileData.replace(/^\uFEFF/, "");
-        const configFile: IConfigurationFile = JSON.parse(fileData);
-        configFile.rulesDirectory = getRulesDirectories(configFile.rulesDirectory, path.dirname(configFilePath));
+        let configFile: IConfigurationFile;
+        let configFileDir: string;
+        if (fs.existsSync(configFilePath)) {
+            let fileData = fs.readFileSync(configFilePath, "utf8");
+            // remove BOM if present
+            fileData = fileData.replace(/^\uFEFF/, "");
+            configFile = JSON.parse(fileData);
+            configFileDir = path.dirname(configFilePath);
+        } else {
+            configFile = require(configFilePath);
+            configFileDir = path.resolve(`./node_modules/${configFilePath}`);
+        }
+
+        configFile.rulesDirectory = getRulesDirectories(configFile.rulesDirectory, configFileDir);
+        configFile.extends = arrayify(configFile.extends);
+
+        for (const name of configFile.extends) {
+            const baseConfigFilePath = resolveConfigurationPath(name, configFileDir);
+            const baseConfigFile = loadConfigurationFromPath(baseConfigFilePath);
+            configFile = extendConfigurationFile(configFile, baseConfigFile);
+        }
         return configFile;
     }
 }
 
+/**
+ * Resolve configuration file path
+ * @var relativeFilePath Relative path or package name (tslint-config-X) or package short name (X)
+ */
+function resolveConfigurationPath(relativeFilePath: string, relativeTo: string) {
```

Comment:
Interesting. I looked at stylelint.  It is using `resolve-from` package. which uses `require('module')`. Do you want to use the `resolve-from` package, or I just re-use those code?